internal/handlers/common/compare.go

```diff
@@ -140,9 +164,12 @@ func compareScalars(a, b any) compareResult {
 		case int64:
 			return compareOrdered(a, b)
 		default:
-			panic(fmt.Sprintf("unexpected type %T", b))
+			return notEqual
 		}
 
+	case *types.Document, *types.Array:
+		return notEqual
```

Comment:
That function is called `compareScalars` so it should compare scalar values. Document and Array are not scalars: https://github.com/FerretDB/FerretDB/blob/17a490df209cd40816003e095fe3db5753a4f527/internal/types/types.go#L28-L44

We should either add another function like `func compare(a, b any) compareResult` that would do compare for documents, arrays, and scalars, or make the caller check for document/array. We should not compare non-scalar values in a function called `compareScalars`

Reply:
How about rename `compareScalars` to `compare`?

Reply:
I think there are cases where we care only about scalars. Put another way, I think there are cases where we should actually compare documents and arrays, and cases where they are not comparable. So let's _add_ a function `compare` that would return `notEqual` for documents and arrays (for now) and call `compareScalars` for other types

Reply:
Agreed.